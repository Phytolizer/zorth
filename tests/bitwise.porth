include "std.porth"

// shift left
1  3 shl print

"------------------------------\n" stdout write

// shift right
32 3 shr print

"------------------------------\n" stdout write

// bit or
1 2 bor print

"------------------------------\n" stdout write

// bit and
1 2 band print
