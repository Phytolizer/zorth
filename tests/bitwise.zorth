// shift left
1 3 shl dump

// shift right
32 3 shr dump

// bit or
1 2 bor dump

// bit and
1 2 band dump
