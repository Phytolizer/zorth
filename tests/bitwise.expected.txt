: returncode 0
: stdout 101
8
------------------------------
4
------------------------------
3
------------------------------
0

: stderr 0

